ivy_tests/test_ivy/test_frontends/test_numpy/test_indexing_routines/test_indexing_like_operations.py

```diff
@@ -0,0 +1,59 @@
+# Testing Function
+# global
+from hypothesis import strategies as st
+
+# local
+import ivy_tests.test_ivy.helpers as helpers
+import ivy_tests.test_ivy.test_frontends.test_numpy.helpers as np_frontend_helpers
+from ivy_tests.test_ivy.helpers import handle_frontend_test
+import ivy
+
+
+@handle_frontend_test(
+    fn_tree="numpy.diagonal",
+    dtype_x_axis=helpers.dtype_values_axis(
+        available_dtypes=helpers.get_dtypes("numeric"),
+        num_arrays=1,
+        shared_dtype=True,
+        min_num_dims=2,
+        min_axis=-2,
+        max_axis=1,
+    ),
+    offset=st.integers(min_value=-1, max_value=1),
+)
+def test_numpy_diagonal(
+    dtype_x_axis,
+    # dtype_and_x,
+    # dtype_and_axis,
+    as_variable,
+    num_positional_args,
+    native_array,
+    with_out,
+    offset,
+    on_device,
+    fn_tree,
+    frontend,
+):
+    input_dtype, x, axis = dtype_x_axis
+    if axis < 0:
+        new_axis = axis + 1
+    else:
+        new_axis = axis - 1
+
+    print("axis", axis)
+    as_variable = as_variable
+
+    np_frontend_helpers.test_frontend_function(
+        input_dtypes=input_dtype,
+        as_variable_flags=as_variable,
+        on_device=on_device,
+        frontend=frontend,
+        fn_tree=fn_tree,
+        with_out=False,
+        num_positional_args=num_positional_args,
+        native_array_flags=native_array,
+        a=ivy.native_array(x, dtype=ivy.int32),
```

Comment:
this should be `a=x[0],`
which then means you should also remove `import ivy`

Reply:
I think the problem is not with the array, it's axes values. Have you run the code?

Reply:
It's generating random tuple and non tuple values including None everytime. I'm pushing the code for your perusal.

Reply:
Yes, I did run the code and it's passing with the changes I requested.

Reply:
I think you forgot to uncomment `min_num_dims=2,`. Also, did you sync with our current master?